### PR TITLE
SER-4349 A/B/C adding drop down on Global Nav with a search scope [mweb]

### PR DESCRIPTION
**Pull request by mmeller-wikia**
https://wikia-inc.atlassian.net/browse/SER-4349

### CODE REVIEW

File: app/components/global-navigation/search.js [hunk 1]
```diff
@@ -69,36 +69,49 @@ export default Component.extend({
 
 	requestSuggestionsFromAPI() {
 		const { query } = this.state;
-		const uri = `${this.get('model.suggestions.url')}&query=${query}`;
-
+		const scope = this.get('searchScope');
```
Comment (latata) [inline app/components/global-navigation/search.js:72]: Let's add default value for scope e.g. `const scope = this.get('searchScope') || '';`
I think that without it we could end up with requests with param `&scope=undefined`

Comment (mmeller-wikia) [reply app/components/global-navigation/search.js]: I set default fallback to this-wiki search

File: dist/scss/wds-components/_global-navigation-search.scss [hunk 2]
```diff
@@ -7,6 +7,20 @@ $toggle-active-margin-right: 17px;
 	height: 100%;
 	justify-content: flex-end;
 
+	&-scope-dropdown {
+		&::before, &::after {
+			z-index: 150 + 1;
+		}
+
+		&::after {
+			bottom: -13px !important;
```
Comment (mmeller-wikia) [inline dist/scss/wds-components/_global-navigation-search.scss:16]: I don't like this `!important` but i lost my patience for this style

File: dist/scss/wds-components/_global-navigation-search.scss [hunk 3]
```diff
@@ -7,6 +7,20 @@ $toggle-active-margin-right: 17px;
 	height: 100%;
 	justify-content: flex-end;
 
+	&-scope-dropdown {
+		&::before, &::after {
+			z-index: 150 + 1;
```
Comment (lomiarz) [inline dist/scss/wds-components/_global-navigation-search.scss:12]: how about to just write 151 🔢

File: app/components/global-navigation/search.js [hunk 4]
```diff
@@ -56,6 +56,17 @@ export default Component.extend({
 
 		// key: query string, value: Array<SearchSuggestionItem>
 		this.cachedResults = {};
+		this.setScopeMessage(this.getScope());
+	},
+
+	setScopeMessage(scope) {
+		if (scope === 'cross-wiki') {
+			this.set('model.placeholder-active.key', 'global-navigation-search-placeholder-all-wikis');
```
Comment (lomiarz) [inline app/components/global-navigation/search.js:64]: small thing, please try to keep consistent names, here you have `cross-wiki` vs `all-wikis`

Comment (lomiarz) [reply app/components/global-navigation/search.js:64]: also you could introduce some enumeration for those hardcoded keys

File: app/components/global-navigation/search.js [hunk 5]
```diff
@@ -377,9 +423,24 @@ export default Component.extend({
 		}
 	},
 
+	getScope() {
+		return this.get('searchScope') === 'cross-wiki' ? 'cross-wiki' : 'internal';
```
Comment (lomiarz) [inline app/components/global-navigation/search.js:427]: what is a benefit from that function ? searchScope can have only two values

File: app/templates/components/global-navigation/search.hbs [hunk 6]
```diff
@@ -84,6 +117,9 @@
                 data-tracking-label={{model.suggestions.tracking-label}}
               >
                 {{{suggestion.text}}}
+                {{#if suggestion.sitename}}
+                  <span style="float: right">{{{suggestion.sitename}}}</span>
```
Comment (lomiarz) [inline app/templates/components/global-navigation/search.hbs:121]: IMHO you could add a class name and describe style in css file

**Review by kejwmen — COMMENTED** [review]
Please take a look at @lomiarz comments

**Review by kejwmen — APPROVED** [review]

**Pull request merged**